### PR TITLE
[specific ci=1-10-Docker-PS] Support docker ps filtering by volume and network

### LINKED ISSUE #3799: container filtering by volumes
`docker ps` allows for container filtering by volume name or volume mount point.  vic currently does not support filtering containers by their volume information.

The container_handler will need to be updated to return the container volume information when requesting a container list.

### LINKED ISSUE #3977: container filtering by multiple networks returns incorrect results
When executing a container list and specifying multiple networks VIC will perform an `and` on the network names instead of an `or` operation.  This incorrect filter operation will result in incorrect results.

Expected (Docker 1.13):
```
docker network ls
NETWORK ID          NAME                DRIVER              SCOPE
c4af6c191fbf        bridge              bridge              local
40c1bc7ee106        development         bridge              local
5c3c4556460d        host                host                local
f17fefbfdf48        none                null                local
74c64208de0b        production          bridge              local

docker run -d --net production --name web_prod busybox /bin/top
d17c5b18af041612a3a474d93fb692644a531f668bad111da7338d1785d84b12

docker run -d --net development --name web_dev busybox /bin/top
d949bbb83895730bdf6d59e6ac6e7f2ea8a59f68ad2c1f6ee569a16a7592cb35

docker ps -f network=production
CONTAINER ID        IMAGE               COMMAND             CREATED              STATUS              PORTS               NAMES
d17c5b18af04        busybox             "/bin/top"          About a minute ago   Up About a minute                       web_prod

docker ps -f network=development
CONTAINER ID        IMAGE               COMMAND             CREATED              STATUS              PORTS               NAMES
d949bbb83895        busybox             "/bin/top"          About a minute ago   Up About a minute                       web_dev

docker ps -f network=development -f network=production
CONTAINER ID        IMAGE               COMMAND             CREATED              STATUS              PORTS               NAMES
d949bbb83895        busybox             "/bin/top"          About a minute ago   Up About a minute                       web_dev
d17c5b18af04        busybox             "/bin/top"          About a minute ago   Up About a minute                       web_prod
```

Actual:
```
docker network ls
NETWORK ID          NAME                DRIVER
d665d230a803        bridge              bridge
2d6454fb7dca        development         bridge
24c40cb8d9c5        production          bridge

docker run -d --net production --name web_prod busybox /bin/top
e43dcce8e75bb16edc934367f93377241225cfae23ade87094c941b030950c3d

docker run -d --net development --name web_dev busybox /bin/top
87056951dab9993b061d343ff6e52a229502e3254c63a2115ee585a3d313fb37

docker ps -f network=production
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
e43dcce8e75b        busybox             "/bin/top"          46 seconds ago      Up 46 seconds                           web_prod

docker ps -f network=development
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
87056951dab9        busybox             "/bin/top"          31 seconds ago      Up 30 seconds                           web_dev

docker ps -f network=development -f network=production
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
```

### PR DESCRIPTION
**Pull request by anchal-agrawal**
Fixes #3799
Fixes #3977

### CODE REVIEW

File: lib/apiservers/engine/backends/filter/container.go [hunk 1]
```diff
@@ -87,31 +84,59 @@ func IncludeContainer(listContext *ContainerListContext, container *models.Conta
...
-				}
-				return fmt.Errorf("not part of container network")
-			})
-			if err != nil {
-				return ExcludeAction
+	if listContext.Filters.Include("network") {
+		netFilterValues := listContext.Filters.Get("network")
+
+		// Gather the container's networks in a map
+		numNetworks := len(container.Endpoints)
```
Comment (cgtexmex) [inline lib/apiservers/engine/backends/filter/container.go]: networks and volumes act as an "or" filter for docker -- so if the container has _any_ of the filter values then it should be **included** in the return.  The current code will exclude containers that should be included in both volume and network filtering.

I suggest you loop over the filters -- as soon as a filter is value found break the loop and that container should be **included** in the results -- if no filter values are found then exclude it.  I would do this without the creation of the volume / network maps -- in both cases just loop over the filters and then loop over the networks / volumes.  I'm expecting there will typically be few filters, so iterating of the collections is fine.

Comment (anchal-agrawal) [inline lib/apiservers/engine/backends/filter/container.go]: I observed this (interesting) behavior with Docker 1.13:

```
$ docker create -v foo:/dir busybox
6adc0f4ce8ced54692b839cd9a640371b0f46fdea4c95d3129dcb128f8656446
$ docker ps -a -f volume=foo
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
6adc0f4ce8ce        busybox             "sh"                16 seconds ago      Created                                 sleepy_clarke
$ docker ps -a -f volume=foo -f network=bar
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
$ docker ps -a -f network=bar -f volume=foo
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
$ docker ps -a -f volume=foo -f volume=bar
CONTAINER ID        IMAGE               COMMAND             CREATED              STATUS              PORTS               NAMES
6adc0f4ce8ce        busybox             "sh"                About a minute ago   Created                                 sleepy_clarke
$ docker ps -a -f volume=foo -f volume=bar -f network=buz
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
```

It looks like `volume` and `network` filters are OR ops by themselves, but if used together they're not and their filter conditions must be satisfied individually.

Comment (cgtexmex) [inline lib/apiservers/engine/backends/filter/container.go]: Agreed -- if individually implemented as OR they it _should_ function in that manner

Comment (anchal-agrawal) [inline lib/apiservers/engine/backends/filter/container.go]: The code in its current form works as expected for the above cases. I've pushed a commit with another integration test. I'll refactor the map to use a list in a later commit.

**Pull request merged**